### ISSUE TITLE
Add support for calling server plugins directly from the client

### ISSUE THREAD

**Issue by Rednas83** [nuxt/nuxt#28584]
I know it can be done with `server/api/*.ts` and `$fetch` in the client but would it not be much more convenient to call `plugins/*.server.ts` directly. Just like with `plugins/*.ts` by using `provide`

### Something like
**pages/hello.vue**
```
<template>
  <h1>Hello world</h1>
  <pre>{{ $hellofn("Hello from server!!") || $helloattr }}</pre> <!-- Returns $setup.$helloconsole is not a function-->
  <pre>{{ $helloconsole("Hello from server!!" || $helloattr) }}</pre> <!-- Returns  ERROR  [nitro] [unhandledRejection] $setup.$helloconsole is not a function-->
</template>

<script setup lang="ts">
const { $hellofn, $helloattr, $helloconsole } = useNuxtApp()
</script>
```

**plugins/hello.server.ts**
```
export default defineNuxtPlugin(() => {
  return {
    provide: {
      hellofn: (msg: string) => `Hello ${msg}!`,      
      helloattr: "Hello from attribute!",
      helloconsole: (msg: string) => console.log(`Hello ${msg}!`),
    },
  }
})
```

### Inspired by
https://github.com/wobsoriano/nuxt-remote-fn which is limited to remote calls only
https://trpc.io/ and https://github.com/wobsoriano/trpc-nuxt which requires lots of boiler plate code but is able to communicate with all RPC servers

**Comment by danielroe** [comment]
We wouldn't recommend this kind of approach (injecting from plugins). Any server function that would be called would face a performance cost (of initiating the entire Vue app instance - required for plugins), and there are already implementations like `nuxt-remote-fn` or https://github.com/antfu/nuxt-server-fn which enable direct function calling.

**Comment by Rednas83** [comment]
Idk. As a workaround I moved the `plugins/*server.ts` to `server/functions` and wrote a `server/api` to be able to acces them from the client.

I know nuxt is mainly about client and ssr but I think there is room for improvement for the server/client communication looking at for example https://trpc.io/

**Some issue I ran into**
* Plugins and server plugins are not auto imported on server level but and must be manually imported. This is only detected in runtime and not by the compiler itself
* Sometimes I am getting `Cannot access 'renderer$1' before initialization` which is quite hard to fix/workaround. Is there perhaps a way to intercept it and make it more readable? See also https://github.com/nuxt/nuxt/issues/20576
![image](https://github.com/user-attachments/assets/1ecc4239-1aa4-4ffe-88e5-d07b2e816b20)
* I also tried `nuxt-remote-fn` but it uses the `lib` folder and is mostly limited to remote calls. See also https://github.com/wobsoriano/nuxt-remote-fn/issues/14
